--- postorius/forms/list_forms.py.orig	2023-04-02 10:59:56.938642619 +0200
+++ postorius/forms/list_forms.py	2023-04-02 11:23:51.923881265 +0200
@@ -38,6 +38,13 @@
 from postorius.models import EmailTemplate, _email_template_help_text
 from postorius.utils import LANGUAGES
 
+# CAPTCHA support
+from django.conf import settings
+from captcha.fields import ReCaptchaField
+from hcaptcha.fields import hCaptchaField
+from friendly_captcha.fields import FrcCaptchaField
+from turnstile.fields import TurnstileField
+
 
 DIGEST_FREQUENCY_CHOICES = (
     ('daily', _('Daily')),
@@ -185,6 +202,17 @@
         label=_('Your name (optional)'), required=False
     )
 
+    # CAPTCHA support
+    captcha_service = getattr(settings, 'CAPTCHA_SERVICE', None)
+    if captcha_service == 'recaptcha':
+        captcha = ReCaptchaField()
+    elif captcha_service == 'hcaptcha':
+        hcaptcha = hCaptchaField()
+    elif captcha_service == 'friendlycaptcha':
+        captcha = FrcCaptchaField()
+    elif captcha_service == 'turnstile':
+        captcha = TurnstileField()
+
     delivery_mode = delivery_mode_field(default='regular')
     delivery_status = delivery_status_field(
         choices=DELIVERY_STATUS_CHOICES, widget=forms.RadioSelect
@@ -216,6 +243,17 @@
         label=_('Your name (optional)'), required=False
     )
 
+    # CAPTCHA support
+    captcha_service = getattr(settings, 'CAPTCHA_SERVICE', None)
+    if captcha_service == 'recaptcha':
+        captcha = ReCaptchaField()
+    elif captcha_service == 'hcaptcha':
+        hcaptcha = hCaptchaField()
+    elif captcha_service == 'friendlycaptcha':
+        captcha = FrcCaptchaField()
+    elif captcha_service == 'turnstile':
+        captcha = TurnstileField()
+
 
 class ListSettingsForm(forms.Form):
     """
