--- django/contrib/admin/forms.py.orig	2023-04-28 23:46:22.384173775 +0200
+++ django/contrib/admin/forms.py	2023-04-30 07:58:37.273713612 +0200
@@ -2,12 +2,30 @@
 from django.core.exceptions import ValidationError
 from django.utils.translation import gettext_lazy as _
 
+# CAPTCHA support
+from django.conf import settings
+from captcha.fields import ReCaptchaField
+from hcaptcha.fields import hCaptchaField
+from friendly_captcha.fields import FrcCaptchaField
+from turnstile.fields import TurnstileField
+
 
 class AdminAuthenticationForm(AuthenticationForm):
     """
     A custom authentication form used in the admin app.
     """
 
+    # CAPTCHA support
+    captcha_service = getattr(settings, 'CAPTCHA_SERVICE', None)
+    if captcha_service == 'recaptcha':
+        captcha = ReCaptchaField()
+    elif captcha_service == 'hcaptcha':
+        captcha = hCaptchaField()
+    elif captcha_service == 'friendlycaptcha':
+        captcha = FrcCaptchaField()
+    elif captcha_service == 'turnstile':
+        captcha = TurnstileField()
+
     error_messages = {
         **AuthenticationForm.error_messages,
         "invalid_login": _(
--- django/contrib/admin/templates/admin/login.html.orig	2023-04-30 08:00:58.881744778 +0200
+++ django/contrib/admin/templates/admin/login.html	2023-04-30 08:04:13.720790267 +0200
@@ -53,6 +53,10 @@
     {{ form.password.label_tag }} {{ form.password }}
     <input type="hidden" name="next" value="{{ next }}">
   </div>
+  <div class="form-row">
+    {{ form.captcha.errors }}
+    {{ form.captcha }}
+  </div>
   {% url 'admin_password_reset' as password_reset_url %}
   {% if password_reset_url %}
   <div class="password-reset-link">
